presto-main/src/test/java/com/facebook/presto/block/TestArrayBlock.java

```diff
@@ -0,0 +1,185 @@
+            if (expectedValues[i] == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                BlockBuilder intermediateBlockBuilder = new ArrayBlockBuilder(BIGINT, new BlockBuilderStatus(), 100, 100);
+                for (int j = 0; j < expectedValues[i].length; j++) {
+                    if (expectedValues[i][j] == null) {
+                        intermediateBlockBuilder.appendNull();
+                    }
+                    else {
+                        BlockBuilder innerMostBlockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), expectedValues[i].length);
+                        for (long v : expectedValues[i][j]) {
+                            BIGINT.writeLong(innerMostBlockBuilder, v);
+                        }
+                        intermediateBlockBuilder.writeObject(innerMostBlockBuilder.build()).closeEntry();
+                    }
+                }
+                blockBuilder.writeObject(intermediateBlockBuilder.build()).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    private void assertArrayValues(long[][] expectedValues)
+    {
+        BlockBuilder blockBuilder = new ArrayBlockBuilder(BIGINT, new BlockBuilderStatus(), 100, 100);
+        for (int i = 0; i < expectedValues.length; i++) {
+            if (expectedValues[i] == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                BlockBuilder elementBlockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), expectedValues[i].length);
+                for (long v : expectedValues[i]) {
+                    BIGINT.writeLong(elementBlockBuilder, v);
+                }
+                blockBuilder.writeObject(elementBlockBuilder).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    private void assertArrayValues(Slice[][] expectedValues)
+    {
+        BlockBuilder blockBuilder = new ArrayBlockBuilder(VARCHAR, new BlockBuilderStatus(), 100, 100);
+        for (int i = 0; i < expectedValues.length; i++) {
+            if (expectedValues[i] == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                BlockBuilder elementBlockBuilder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), expectedValues[i].length);
+                for (Slice v : expectedValues[i]) {
+                    VARCHAR.writeSlice(elementBlockBuilder, v);
+                }
+                blockBuilder.writeObject(elementBlockBuilder.build()).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    @Override
+    protected <T> void assertPositionValue(Block block, int position, T expectedValue)
```

Comment:
I don't really like the idea of making this a member function, just to override it in this one class. Instead I would merge this functionality into the original code, by adding another case for `classT == Slice.class`, and probably a helper method that does all the `Slice` checks in the original code, since they're rather extensive